pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java

```
@@ -84,14 +85,7 @@ public String getData(
 
     ZNRecord znRecord = _pinotHelixResourceManager.readZKData(path);
     if (znRecord != null) {
-      byte[] serializeBytes = _znRecordSerializer.serialize(znRecord);
-      if (GZipCompressionUtil.isCompressed(serializeBytes)) {
-        try {
-          serializeBytes = GZipCompressionUtil.uncompress(new ByteArrayInputStream(serializeBytes));
-        } catch (IOException e) {
-          throw new RuntimeException(e);
-        }
-      }
+      byte[] serializeBytes = serialize(znRecord);
       return new String(serializeBytes, StandardCharsets.UTF_8);
     }
     return null;
```

Should throw `NOT_FOUND` instead of returning `null`